comictaggerlib/ui/talkeruigenerator.py

```diff
@@ -0,0 +1,239 @@
+    widget = QtWidgets.QCheckBox(option.display_name)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, layout.rowCount(), 0, 1, -1)
+
+    return widget
+
+
+def generate_spinbox(option: settngs.Setting, layout: QtWidgets.QGridLayout) -> QtWidgets.QSpinBox:
+    row = layout.rowCount()
+    lbl = QtWidgets.QLabel(option.display_name)
+    layout.addWidget(lbl, row, 0)
+    widget = QtWidgets.QSpinBox()
+    widget.setRange(0, 9999)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, row, 1, alignment=QtCore.Qt.AlignLeft)
+
+    return widget
+
+
+def generate_doublespinbox(option: settngs.Setting, layout: QtWidgets.QGridLayout) -> QtWidgets.QDoubleSpinBox:
+    row = layout.rowCount()
+    lbl = QtWidgets.QLabel(option.display_name)
+    layout.addWidget(lbl, row, 0)
+    widget = QtWidgets.QDoubleSpinBox()
+    widget.setRange(0, 9999.99)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, row, 1, alignment=QtCore.Qt.AlignLeft)
+
+    return widget
+
+
+def generate_textbox(
+    option: settngs.Setting, layout: QtWidgets.QGridLayout
+) -> tuple[QtWidgets.QLineEdit, QtWidgets.QPushButton]:
+    btn = None
+    row = layout.rowCount()
+    lbl = QtWidgets.QLabel(option.display_name)
+    layout.addWidget(lbl, row, 0)
+    widget = QtWidgets.QLineEdit()
+    widget.setObjectName(option.internal_name)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, row, 1)
+
+    # Special case for api_key, make a test button
+    if option.internal_name.endswith("api_key"):
+        btn = QtWidgets.QPushButton("Test Key")
+        layout.addWidget(btn, row, 2)
+
+    if option.internal_name.endswith("url"):
+        btn = QtWidgets.QPushButton("Test URL")
+        layout.addWidget(btn, row, 2)
+
+    return widget, btn
+
+
+def settings_to_talker_form(sources: dict[str, QtWidgets.QWidget], config: settngs.Config[settngs.Namespace]) -> None:
+    # Set the active talker in sources combo box
+    sources["talker_source"].setCurrentIndex(sources["talker_source"].findData(config[0].talker_source))
+
+    for talker in sources["tabs"].items():
+        for name, widget in talker[1]["widgets"].items():
+            value = getattr(config[0], name)
+            value_type = type(value)
+            try:
+                if value_type is str:
+                    widget.setText(value)
+                if value_type is int or value_type is float:
+                    widget.setValue(value)
+                if value_type is bool:
+                    widget.setChecked(value)
+            except Exception:
+                logger.debug("Failed to set value of %s", name)
+
+
+def form_settings_to_config(sources: dict[str, QtWidgets.QWidget], config: settngs.Config[settngs.Namespace]) -> None:
+    # Source combo box value
+    config[0].talker_source = sources["talker_source"].itemData(sources["talker_source"].currentIndex())
```

Comment:
This can be a single call to `currentText`.

Reply:
That will return the name (Comic Vine) and not the talker_id (comicvine) which is in the item data and is what needs to be saved to config.

Reply:
Looks like there is currentData() is that what you meant?